diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2fb186a59..333fa09a4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,9 +66,10 @@ install(TARGETS ${BINARY_NAME} RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}"
 file(GLOB FLUTTER_PLUGIN_DIRS "${PROJECT_BINARY_DIR}/plugins/*")
 
 INSTALL(CODE "
+  execute_process(COMMAND patchelf --set-rpath $ORIGIN ${FLUTTER_LIBRARY_DIR}/libflutter_linux_glfw.so)
   include(BundleUtilities)
   fixup_bundle(\"\${CMAKE_INSTALL_PREFIX}/${BINARY_NAME}\" \"\"
-    \"${FLUTTER_PLUGIN_DIRS};${FLUTTER_LIBRARY_DIR}\")
+    \"${FLUTTER_PLUGIN_DIRS};${FLUTTER_LIBRARY_DIR};/usr/lib/x86_64-linux-gnu;/snap/flutter/current/usr/lib/x86_64-linux-gnu\")
   " COMPONENT Runtime)
 
 INSTALL(FILES "${FLUTTER_ICU_DATA_FILE}" DESTINATION "${INSTALL_BUNDLE_DATA_DIR}"
